src/validation_workflow/docker/validation_docker.py

```diff
@@ -0,0 +1,203 @@
+        # replace the placeholder in template docker-compose.yaml which is taken from sample in https://opensearch.org/docs/latest/opensearch/install/docker/#sample-docker-composeyml
+        yml_template = 'docker-compose.yml'
+        yml_template_location = './src/validation_workflow/docker/'
+
+        tmp_dir = TemporaryDirectory()
+        shutil.copy2(yml_template_location + yml_template, tmp_dir.name)
+        target_yml_file = tmp_dir.name + '/' + yml_template
+        self.inplace_change(target_yml_file, 'OS_IMAGE_PLACEHOLDER', OpenSearch_image + ":" + version)
+        self.inplace_change(target_yml_file, 'OSD_IMAGE_PLACEHOLDER', OpenSearchDashboard_image + ":" + version)
+        self.inplace_change(target_yml_file, 'OS1_NAME', OpenSearchNode1_name)
+        self.inplace_change(target_yml_file, 'OS2_NAME', OpenSearchNode2_name)
+        self.inplace_change(target_yml_file, 'OSD_NAME_PLACEHOLDER', OpenSearchDashboard_name)
+
+        # spin up containers
+        docker_compose_up = 'docker-compose -f ' + target_yml_file + ' up -d'
+        result = subprocess.run(docker_compose_up, shell=True, stdout=PIPE, stderr=PIPE, universal_newlines=True)
+
+        return ('returncode=0' in (str(result)), target_yml_file)
+
+    def inplace_change(self, filename: str, old_string: str, new_string: str) -> None:
+
+        with open(filename) as f:
+            s = f.read()
+            if old_string not in s:
+                print('"{old_string}" not found in {filename}.'.format(**locals()))
+                return
+
+        with open(filename, 'w') as f:
+            print('Changing "{old_string}" to "{new_string}" in {filename}'.format(**locals()))
+            s = s.replace(old_string, new_string)
+            f.write(s)
+
+    def get_image_id(self, image_name: str, image_version: str) -> str:
+        local_inspect = "docker image inspect -f '{{ .Id }}' " + image_name + ":" + image_version
+        local_remove = "docker image rm -f " + image_name + ":" + image_version
+
+        # Removing the image if it exists at local
+        result_inspect = subprocess.run(local_inspect, shell=True, stdout=PIPE, stderr=PIPE, universal_newlines=True)
+        if (result_inspect.returncode == 0):
+            logging.info('Image exists at local : ' + result_inspect.stdout + ' : ' + image_name + ":" + image_version)
+            logging.info('removing the local image and pulling a fresh one from DockerHub')
+            result_remove = subprocess.run(local_remove, shell=True, stdout=PIPE, stderr=PIPE, universal_newlines=True)
+            if (result_remove.returncode == 0):
+                logging.info('Image is removed at local : ' + image_name + ":" + image_version)
+            else:
+                raise Exception('Image removal fail : return code ' + str(result_remove.returncode))
+        else:
+            logging.info('Image does not exist at local, proceed with Pull from the dockerHub')
```

Comment:
Can you check if you force pull instead of removing and then checking locally? Force pull will override the local image as well

Reply:
Thanks, @gaiksaya , let me use '-f' to simplify the code.. thanks again... I now recall @prudhvigodithi said the same recommendation in the dry-run as well.. :)